pkg/core/resources/apis/mesh/mesh_validator.go

```diff
@@ -264,6 +265,10 @@ func validatePrometheusConfig(cfgStr *structpb.Struct) validators.ValidationErro
 		verr.AddViolation("", fmt.Sprintf("could not parse config: %s", err.Error()))
 		return verr
 	}
+	if _, err := regexp.Compile(cfg.GetEnvoy().GetFilterRegex()); err != nil {
```

Comment:
Envoy uses `re2` regex, have you checked the syntax is the same?

Reply:
> The syntax of the regular expressions accepted is the same general syntax used by Perl, Python, and other languages. More precisely, it is the syntax accepted by RE2 and described at https://golang.org/s/re2syntax, except for \C. 
https://pkg.go.dev/regexp